src/UI/WhiteBoardController.java

```diff
@@ -0,0 +1,107 @@
+package UI;
+import infrastructure.validation.logger.*;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.util.Base64;
+import javax.imageio.ImageIO;
+import processing.*;
+
+/***
+ * @Author	 : Shiva Dhanush
+ * This controller handles operations on going from startsession page to canvas page.
+ * It takes the image selected by user and encodes it into a base64 string.
+ * It sends user details to processing module and content module.
+ ***/
+
+import org.json.simple.JSONObject;
+import infrastructure.content.*;
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.stage.FileChooser;
+import javafx.stage.Stage;
+public class WhiteBoardController {
+   
+	@FXML
+	private TextField userName;
+	@FXML
+	private TextField ipAddress;
+	@FXML
+	private TextField boardId;
+	@FXML
+	private Button imageChoose;
+	private String encodedImage;
+	static ILogger logger = LoggerFactory.getLoggerInstance();
+	
+	/**
+	 *This function takes jpg image and converts it to base64 string.
+	 *@param event The event is selecting image.
+	 *@returns nothing
+	 */
+	public void getImage (ActionEvent event) throws Exception
+	{
+	try {
+		/** This method takes in the image of jpg type and encodes it into a string. */
+		FileChooser fc = new FileChooser();
+		File f = fc.showOpenDialog(null);
+		if(f!=null)
+		{   
+			/** Byte array of image file taken as input is encoded with Base64. */
+			BufferedImage originalImage = ImageIO.read(f);
+			ByteArrayOutputStream baos = new ByteArrayOutputStream();
+			ImageIO.write( originalImage, "jpg", baos );
+			baos.flush();
+			byte[] imagebyte = baos.toByteArray();
+			encodedImage = Base64.getEncoder().encodeToString(imagebyte);
+			baos.close();
+		}
+		}
+	catch(Exception e){
+		logger.log(ModuleID.UI, LogLevel.ERROR, "Error in encoding image");
+		}
+	}
+	
+	/**
+	 *This function sends data to processing,content moudles and opens canvas.
+	 *@param event The event is clicking start session button.
+	 *@returns nothing
+	 */
+	public void startSession (ActionEvent event) throws Exception
+	{   
+	try {
+		/** Sending the data filled to the processing module */
+		Processor processor = ProcessingFactory.getProcessor() ;
+		IUser user = processor;  
+		String returnval= user.giveUserDetails(userName.getText(),ipAddress.getText(),boardId.getText());
```

Comment:
We are returning the boardID here, if user is asking for a new board, he will give boardID as empty string or null value, then we will set the boardID and return to you. So you should save it.

Reply:
@sai-deekshith16 is this taken care of?

Reply:
@himanshu2047 i think it is needed only for more than one session support right>

Reply:
More than one session support is there @muhammedyaseen-git